src/doc_table.c

```diff
@@ -188,125 +233,181 @@ void dmd_free(RSDocumentMetadata *md) {
   }
   sdsfree(md->keyPtr);
 }
+
 void DocTable_Free(DocTable *t) {
-  // we start at docId 1, not 0
-  for (int i = 1; i < t->size; i++) {
-    dmd_free(&t->docs[i]);
-  }
-  if (t->docs) {
-    rm_free(t->docs);
+  for (int i = 0; i < t->cap; ++i) {
+    DMDChain *chain = &t->buckets[i];
+    if (DMDChain_IsEmpty(chain)) {
+      continue;
+    }
+    RSDocumentMetadata *md = chain->first;
+    while (md) {
+      dmd_free(md);
+      md = md->next;
```

Comment:
if the dmd is freed, this line will be invalid access. you need to assign a temporary next variable.

Reply:
If the dmd is freed it removed from the chain

Reply:
what I mean is that line 245 should free the MD (it doesn't do so currently, but it *should*). If that's the case, line 246 will cause invalid mem access. Something like:

```c
DMD *next = md->next;
dmd_free(md);
md = next;
```